tools/semgrep-py.yml

```diff
@@ -36,6 +36,7 @@ rules:
       - pattern-not: Message.objects.filter(..., id__in=..., ...)
       - pattern-not: Message.objects.filter(..., id__lt=..., ...)
       - pattern-not: Message.objects.filter(..., id__gt=..., ...)
+      - pattern-not: Message.objects.filter(date_sent__gte=date_days_ago, recipient__type=Recipient.STREAM)
```

Comment:
No, this is not a valid exception.  Looking forward to when we start sharding messages by realm, we will _need_ all queries to be realm-limited.  We should not start making more work for ourselves -- add a loop to make this query go realm-by-realm.  We have an index on `(realm_id, recipient_id, date_sent)` so it will still be performant.

This is particularly important because you're running it all in _one_ transaction right now.  Breaking it up into one transaction per realm will make it much less likely to cause problems.

Reply:
Ah! I see. Modified to run per realm.